### ISSUE TITLE
[3.5.0-RC3] GraphQL - Running out of memory when using query variables for 'children'.

### ISSUE THREAD

**Issue by samput** [craftcms/cms#6463]
### Description
After the update to 3.5.0-RC3 I'm running out of memory on a query when using query variables on 'children'. Getting these errors below originating from ArrayHelper.php / BaseArrayHelper.php

```log
[27-Jul-2020 10:26:31 UTC] PHP Fatal error:  Allowed memory size of 536870912 bytes exhausted (tried to allocate 20480 bytes) in /var/www/vendor/yiisoft/yii2/helpers/BaseArrayHelper.php on line 94
[27-Jul-2020 10:26:31 UTC] PHP Fatal error:  Allowed memory size of 536870912 bytes exhausted (tried to allocate 20480 bytes) in /var/www/vendor/opis/closure/src/SerializableClosure.php on line 381
[27-Jul-2020 10:28:09 UTC] PHP Fatal error:  Allowed memory size of 536870912 bytes exhausted (tried to allocate 262144 bytes) in /var/www/vendor/craftcms/cms/src/helpers/ArrayHelper.php on line 50
[27-Jul-2020 10:28:09 UTC] PHP Fatal error:  Allowed memory size of 536870912 bytes exhausted (tried to allocate 262144 bytes) in Unknown on line 0
```
When hardcoding the the values the query runs with no issue.

### Steps to reproduce
I have a structure set up like so:
```
- Year (entryType: year)
    - Quarter (entryType: quarter)
        - Platform (entryType: platform)
            - Regular (entryType: regular)
            - Thematic (entryType: thematic)
```

#### Query (using query variables for children)
```graphql
query Regular($year: [String], $quarter: [String], $platform: [String]) {
  pipeline: entries(section: "pipeline", type: "year", slug: $year) {
    id
    title
    slug
    quarters: children(type: "quarter", slug: $quarter) {
      id
      title
      slug
      platforms: children(type: "platform", slug: $platform) {
        id
        title
        slug
        regular: children(type: "regular") {
          id
          title
          slug
          ...on pipeline_regular_Entry {
            series {
              id
              title
              slug
            }
          }
        }
      }
    }
  }
}
```

#### Query variables
```json
{
  "year": "2020",
  "quarter": "q1",
  "platform": "platform-name"
}
```

#### Result
```json
{
  "name": "PHP Fatal Error",
  "message": "Allowed memory size of 536870912 bytes exhausted (tried to allocate 20480 bytes)",
  "code": 1,
  "type": "yii\\base\\ErrorException",
  "file": "/var/www/vendor/yiisoft/yii2/helpers/BaseArrayHelper.php",
  "line": 94,
  "stack-trace": [
    "#0 [internal function]: yii\\base\\ErrorHandler->handleFatalError()",
    "#1 {main}"
  ]
}
```

with error in phperrors.log
```
[27-Jul-2020 10:26:31 UTC] PHP Fatal error:  Allowed memory size of 536870912 bytes exhausted (tried to allocate 20480 bytes) in /var/www/vendor/yiisoft/yii2/helpers/BaseArrayHelper.php on line 94
[27-Jul-2020 10:26:31 UTC] PHP Fatal error:  Allowed memory size of 536870912 bytes exhausted (tried to allocate 20480 bytes) in /var/www/vendor/opis/closure/src/SerializableClosure.php on line 381
```

or 

#### Empty Result
```json
""
```

with error in phperrors.log
```
[27-Jul-2020 10:28:09 UTC] PHP Fatal error:  Allowed memory size of 536870912 bytes exhausted (tried to allocate 262144 bytes) in /var/www/vendor/craftcms/cms/src/helpers/ArrayHelper.php on line 50
[27-Jul-2020 10:28:09 UTC] PHP Fatal error:  Allowed memory size of 536870912 bytes exhausted (tried to allocate 262144 bytes) in Unknown on line 0
```

#### Working query (hardcoded values for children)
```graphql
query Regular($year: [String]) {
  pipeline: entries(section: "pipeline", type: "year", slug: $year) {
    id
    title
    slug
    quarters: children(type: "quarter", slug: "q1") {
      id
      title
      slug
      platforms: children(type: "platform", slug: "platform-name") {
        id
        title
        slug
        regular: children(type: "regular") {
          id
          title
          slug
          ...on pipeline_regular_Entry {
            series {
              id
              title
              slug
            }
          }
        }
      }
    }
  }
}
```

#### Query variables
```json
{
  "year": "2020"
}
```

#### Result
```json
{
  "data": {
    "pipeline": [
      {
        "id": "297",
        "title": "2020",
        "slug": "2020",
        "quarters": [
          {
            "id": "14",
            "title": "Q1",
            "slug": "q1",
            "platforms": [
              {
                "id": "136",
                "title": "Platform Name",
                "slug": "platform-name",
                "regular": [
                  {
                    "id": "155",
                    "title": "Regular - Platform Name - Q1 - 2020",
                    "slug": "regular-platform-name-q1-2020",
                    "series": [
                      {
                        "id": "440",
                        "title": "How To",
                        "slug": "how-to"
                      }
                    ]
                  }
                ]
              }
            ]
          }
        ]
      }
    ]
  }
}
```

### Additional info

- Craft version: 3.5.0-RC3
- PHP version: 7.3.15
- Database driver & version: MySQL 5.7.24
- Plugins & versions:
    - Amazon S3: 1.2.9
    - Feed Me: 4.3.3
    - Redactor: 2.6.1

**Comment by brandonkelly** [comment]
Can you please send a database backup and your Composer files into support@craftcms.com so we can look into this?

**Comment by samput** [comment]
@brandonkelly sent, thanks for taking a look.

**Comment by brandonkelly** [comment]
Thanks! We’re able to reproduce with those files, and will get this fixed.

**Comment by andris-sevcenko** [comment]
Just pushed a fix for this for the next 3.5 release. Sorry about the dum-dum mistake :)

**Comment by samput** [comment]
@andris-sevcenko Thanks for the quick fix!

**Comment by brandonkelly** [comment]
Craft 3.5 RC4 is out now with the fix.

**Comment by samput** [comment]
Just updated to 3.5.0-RC4 I'm no longer getting the memory error, but I'm now getting `Undefined index: slug` when using a query variable for children 'slug'.

#### Query with variable
```graphql
query Regular($year: [String], $quarter: [String]) {
  pipeline: entries(section: "pipeline", type: "year", slug: $year) {
    id
    title
    slug
    children(type: "quarter", slug: $quarter) {
      id
      title
      slug
    }
  }
}
```
#### Query variables
```json
{
  "year":"2020",
  "quarter":"q1"
}
```

#### Result
```json
{
  "errors": [
    {
      "debugMessage": "Undefined index: slug",
      "message": "Internal server error",
      "category": "internal",
      "locations": [
        {
          "line": 2,
          "column": 3
        }
      ],
      "path": [
        "pipeline"
      ]
    }
  ],
  "data": {
    "pipeline": null
  }
}
```

#### Query without variable
```graphql
query Regular($year: [String]) {
  pipeline: entries(section: "pipeline", type: "year", slug: $year) {
    id
    title
    slug
    children(type: "quarter", slug: "q1") {
      id
      title
      slug
    }
  }
}
```
#### Result
```json
{
  "data": {
    "pipeline": [
      {
        "id": "297",
        "title": "2020",
        "slug": "2020",
        "children": [
          {
            "id": "14",
            "title": "Q1",
            "slug": "q1"
          }
        ]
      }
    ]
  }
}
```

**Comment by andris-sevcenko** [comment]
@samput seeing a few things our testing has missed. Fixing this now.

**Comment by andris-sevcenko** [comment]
Okay! Totally fixed this for the next Craft 3.5 release this time! :)

**Comment by brandonkelly** [comment]
Craft 3.5 RC6 is out now with this fix.
